src/optimize.ts

```diff
@@ -280,7 +292,54 @@ async function processImage(
   /*
    * Check for external images
    */
-  if (!isLocal(attrib_src)) {
+  if (
+    !isLocal(attrib_src) &&
+    !!config.image.external.src_include &&
+    attrib_src.match(config.image.external.src_include) &&
+    (!config.image.external.src_exclude ||
+      !attrib_src.match(config.image.external.src_exclude))
+  ) {
+    switch (config.image.cdn.process) {
+      case 'off':
+        break;
+      case 'optimize':
+        const canonical = getCanonicalCdnForUrl(attrib_src);
+        if (!canonical) break;
+        const cdnTransformer = getTransformer(canonical.cdn);
+        if (!cdnTransformer) break;
+        let attrib_width = getIntAttr(img, 'width');
+        if (!attrib_width) {
+          $state.reportIssue(htmlfile, {
+            type: 'warn',
+            msg: `Missing or malformed'width' attribute for image ${attrib_src}, unable to perform CDN transform`,
+          });
+          return;
+        }
+        let attrib_height = getIntAttr(img, 'height');
+        if (!attrib_height) {
+          $state.reportIssue(htmlfile, {
+            type: 'warn',
+            msg: `Missing or malformed 'height' attribute for image ${attrib_src}, unable to perform CDN transform`,
+          });
+          return;
+        }
+        const new_srcset = await generateSrcSetForCdn(
+          attrib_src,
+          cdnTransformer,
+          attrib_width,
+          attrib_height
+        );
+
+        if (new_srcset !== null) {
+          img.attr('srcset', new_srcset);
+        }
+
+        // Add sizes attribute if not specified
```

Comment:
This piece of code should be later in the process as it's also needed for local images I think.

Reply:
Because much/most of the codeflow after this is specific to local images I duplicated srcSet generation for cdn optimization as it felt simpler this way
It's still done for local images